managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -285,6 +285,37 @@ public MarkDeleteEntry(PositionImpl newPosition, Map<String, Long> properties,
         return lastMarkDeleteEntry != null ? lastMarkDeleteEntry.properties : Collections.emptyMap();
     }
 
+    @Override
+    public boolean putProperty(String key, Long value) {
+        if (lastMarkDeleteEntry != null) {
+            MarkDeleteEntry currentLastMarkDeleteEntry = lastMarkDeleteEntry;
+            Map<String, Long> properties = currentLastMarkDeleteEntry.properties;
+            if (properties == null || properties.isEmpty()) {
```

Comment:
do we have concurrency issues here ?
there is no lock or other synchronisation primitive that prevents two calls to access this variable, the same happens for  `lastMarkDeleteEntry`

Reply:
Fixed the method to update `lastMarkDeleteEntry` using `AtomicReferenceFieldUpdater`.